rx_msg_extras.go

```diff
@@ -261,3 +328,274 @@ func (r *rxLinkMessageSpecifics) GetURL() string {
+
+	// GetState 添加此用户的「联系我」方式配置的state参数，可用于识别添加此用户的渠道
+	GetState() string
+}
+
+var _ EventAddHalfExternalContact = (*rxEventAddHalfExternalContact)(nil)
+
+func (r *rxEventAddHalfExternalContact) formatInto(w io.Writer) {
+	_, _ = fmt.Fprintf(
+		w,
+		"UserID: %#v, ExternalUserID: %#v, State: %#v, WelcomeCode: %#v",
+		r.UserID,
+		r.ExternalUserID,
+		r.State,
+		r.WelcomeCode,
+	)
+}
+
+func (r *rxEventAddHalfExternalContact) GetUserID() string {
+	return r.UserID
+}
+
+func (r *rxEventAddHalfExternalContact) GetExternalUserID() string {
+	return r.ExternalUserID
+}
+
+func (r *rxEventAddHalfExternalContact) GetState() string {
+	return r.State
+}
+
+func (r *rxEventAddHalfExternalContact) GetWelcomeCode() string {
+	return r.WelcomeCode
+}
+
+// EventDelExternalContact 删除企业客户事件
+type EventDelExternalContact interface {
+	messageKind
+
+	// GetUserID 企业服务人员的UserID
+	GetUserID() string
+
+	// GetExternalUserID 外部联系人的userid，注意不是企业成员的帐号
+	GetExternalUserID() string
+}
+
+var _ EventDelExternalContact = (*rxEventDelExternalContact)(nil)
+
+func (r *rxEventDelExternalContact) formatInto(w io.Writer) {
+	_, _ = fmt.Fprintf(
+		w,
+		"UserID: %#v, ExternalUserID: %#v",
+		r.UserID,
+		r.ExternalUserID,
+	)
+}
+
+func (r *rxEventDelExternalContact) GetUserID() string {
+	return r.UserID
+}
+
+func (r *rxEventDelExternalContact) GetExternalUserID() string {
+	return r.ExternalUserID
+}
+
+// EventDelFollowUser 删除跟进成员事件
+type EventDelFollowUser interface {
+	messageKind
+
+	// GetUserID 企业服务人员的UserID
+	GetUserID() string
+
+	// GetExternalUserID 外部联系人的userid，注意不是企业成员的帐号
+	GetExternalUserID() string
+}
+
+var _ EventDelFollowUser = (*rxEventDelFollowUser)(nil)
+
+func (r *rxEventDelFollowUser) formatInto(w io.Writer) {
+	_, _ = fmt.Fprintf(
+		w,
+		"UserID: %#v, ExternalUserID: %#v",
+		r.UserID,
+		r.ExternalUserID,
+	)
+}
+
+func (r *rxEventDelFollowUser) GetUserID() string {
+	return r.UserID
+}
+
+func (r *rxEventDelFollowUser) GetExternalUserID() string {
+	return r.ExternalUserID
+}
+
+// EventTransferFail 客户接替失败事件
+type EventTransferFail interface {
+	messageKind
+
+	// GetUserID 企业服务人员的UserID
+	GetUserID() string
+
+	// GetExternalUserID 外部联系人的userid，注意不是企业成员的帐号
+	GetExternalUserID() string
+
+	// GetFailReason 接替失败的原因, customer_refused-客户拒绝， customer_limit_exceed-接替成员的客户数达到上限
+	GetFailReason() string
+}
+
+var _ EventTransferFail = (*rxEventTransferFail)(nil)
+
+func (r *rxEventTransferFail) formatInto(w io.Writer) {
+	_, _ = fmt.Fprintf(
+		w,
+		"UserID: %#v, ExternalUserID: %#v, FailReason: %#v",
+		r.UserID,
+		r.ExternalUserID,
+		r.FailReason,
+	)
+}
+
+func (r *rxEventTransferFail) GetUserID() string {
+	return r.UserID
+}
+
+func (r *rxEventTransferFail) GetExternalUserID() string {
+	return r.ExternalUserID
+}
+
+func (r *rxEventTransferFail) GetFailReason() string {
+	return r.FailReason
+}
+
+// EventChangeExternalChat 客户群变更事件
+type EventChangeExternalChat interface {
+	messageKind
+
+	// GetChatId 群ID
+	GetChatId() string
```

Comment:
In this library the `ChatId` field name is uniformly spelled like `ChatID`. Please fix the casing.